ProcessAPI/Sources.java

```diff
@@ -0,0 +1,3 @@
+public enum Sources {
+     LOCAL_FILE, NETWORK_DRIVE, DATABASE;

```

Comment:
You'll probably end up with a wrapper around this enum to include type-specific information (filename, network path, etc) - it's probably worth keeping an explicit type enum for ease of serialization, and then having an interface with several implementation (ex: interface InputConfig { public Sources getSourceType(); }, with FileInputConfig, NetworkInputConfig, etc all implementing the interface and containing the specific details for each).